src/core/vdom/helpers/resolve-async-component.js

```diff
@@ -51,18 +51,27 @@ export function resolveAsyncComponent (
     return factory.resolved
   }
 
+  const owner = currentRenderingInstance
+  if (isDef(factory.owners) && factory.owners.indexOf(owner) === -1) {
+    // already pending
+    factory.owners.push(owner)
+  }
+
   if (isTrue(factory.loading) && isDef(factory.loadingComp)) {
     return factory.loadingComp
   }
 
-  const owner = currentRenderingInstance
-  if (isDef(factory.owners)) {
-    // already pending
-    factory.owners.push(owner)
-  } else {
+  if (!isDef(factory.owners)) {
     const owners = factory.owners = [owner]
     let sync = true
 
+    const removeOwner = (destroyedOwner) => {
+      const index = owners.indexOf(destroyedOwner)
+      if (index > -1) owners.splice(index, 1)
+    }
+
+    if (owner) owner.$on('hook:destroyed', () => removeOwner(owner))
+
```

Comment:
This stuff isn't mandatory. It doesn't break anything if destroyed owners are fore updated. But it might be cleaner to remove them.